rmc/src/lib.rs

```diff
@@ -318,158 +318,29 @@ impl<'a, H: Signable + Hash + Eq + Clone + Debug, MK: MultiKeychain> ReliableMul
-
-        fn verify(&self, msg: &[u8], sgn: &Self::Signature, index: NodeIndex) -> bool {
-            self.key_box.verify(msg, sgn, index)
-        }
-    }
-
-    impl<KB: KeyBox> MultiKeychain for DefaultMultiKeychain<KB> {
-        type PartialMultisignature = SignatureSet<KB::Signature>;
-
-        fn from_signature(
-            &self,
-            signature: &Self::Signature,
-            index: NodeIndex,
-        ) -> Self::PartialMultisignature {
-            SignatureSet::add_signature(
-                SignatureSet::with_size(self.node_count()),
-                signature,
-                index,
-            )
-        }
-
-        fn is_complete(&self, msg: &[u8], partial: &Self::PartialMultisignature) -> bool {
-            let signature_count = partial.iter().count();
-            if signature_count < self.quorum() {
-                return false;
-            }
-            partial
-                .iter()
-                .all(|(i, sgn)| self.key_box.verify(msg, sgn, i))
-        }
-    }
-
-    #[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
-    struct TestSignature {
-        msg: Vec<u8>,
-        index: NodeIndex,
-    }
-
-    #[derive(Clone, Debug)]
-    struct TestKeyBox {
-        count: NodeCount,
-        index: NodeIndex,
-    }
-
-    impl TestKeyBox {
-        fn new(count: NodeCount, index: NodeIndex) -> Self {
-            TestKeyBox { count, index }
-        }
-    }
-
-    impl Index for TestKeyBox {
-        fn index(&self) -> NodeIndex {
-            self.index
-        }
-    }
-
-    #[async_trait]
-    impl KeyBox for TestKeyBox {
-        type Signature = TestSignature;
-
-        fn node_count(&self) -> NodeCount {
-            self.count
-        }
-
-        async fn sign(&self, msg: &[u8]) -> Self::Signature {
-            TestSignature {
-                msg: msg.to_vec(),
-                index: self.index,
-            }
-        }
-
-        fn verify(&self, msg: &[u8], sgn: &Self::Signature, index: NodeIndex) -> bool {
-            index == sgn.index && msg == sgn.msg
-        }
-    }
-
-    type TestMultiKeychain = DefaultMultiKeychain<TestKeyBox>;
+    use std::{collections::HashMap, pin::Pin, time::Duration};
 
-    type TestPartialMultisignature = SignatureSet<TestSignature>;
+    type TestMessage = Message<Signable, Signature, PartialMultisignature>;
+    type TestMultiKeychain = Keychain;
+    type TestBadMultiKeychain = BadSignatureWrapper<Keychain>;
 
-    fn test_multi_keychain(node_count: NodeCount, index: NodeIndex) -> TestMultiKeychain {
-        let key_box = TestKeyBox::new(node_count, index);
-        DefaultMultiKeychain::new(key_box)
-    }
-
-    #[derive(Clone, Copy, PartialEq, Eq, Debug, Hash, Ord, PartialOrd)]
-    struct Hash {
-        byte: u8,
-    }
-
-    impl Signable for Hash {
-        type Hash = [u8; 1];
-
-        fn hash(&self) -> Self::Hash {
-            [self.byte]
-        }
+    fn prepare_keychains(node_count: NodeCount) -> Vec<TestMultiKeychain> {
```

Comment:
Hm, this is used enough that perhaps it could be in `mock` directly as a constructor?

Reply:
moved to `Keychain::new_vec`